dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -696,29 +696,58 @@ private SourceCodeRepoInterface getSourceCodeRepoInterface(String gitUrl, User u
     @ApiOperation(value = "Get the DAG for a given workflow version", notes = "", response = String.class)
     public String getWorkflowDag(@ApiParam(value = "workflowId", required = true) @PathParam("workflowId") Long workflowId,
             @ApiParam(value = "workflowVersionId", required = true) @PathParam("workflowVersionId") Long workflowVersionId)  {
-
-        // This will make it easier in the future to add new types (also better coding practice)
         Workflow workflow = workflowDAO.findById(workflowId);
-        Set<WorkflowVersion> workflowVersions = workflow.getVersions();
-        WorkflowVersion workflowVersion = null;
+        WorkflowVersion workflowVersion = getWorkflowVersion(workflow, workflowVersionId);
+        SourceFile mainDescriptor = getMainDescriptorFile(workflowVersion);
+        String result = null;
 
-        for (WorkflowVersion wv : workflowVersions) {
-            if (wv.getId() == workflowVersionId) {
-                workflowVersion = wv;
-                break;
+        if(mainDescriptor != null) {
+            File tmpDir = Files.createTempDir();
+            File tempMainDescriptor = null;
+            try {
+                // Write main descriptor to file
+                // The use of temporary files is not needed here and might cause new problems
+                tempMainDescriptor = File.createTempFile("main", "descriptor", tmpDir);
```

Comment:
Please change this to store and read files from memory rather from disk. 
That would avoid the need to catch an IOException (which this code just throws away anyway) and avoid the possibility of filling up tmp with files.